Migrate/analytics coffee

Pasindufdo98 opened the pull request:
Frontend Migration

# Description

In this PR, I have migrated and removed the old analytics.coffee file, which included the virtualPageview(false) logic and previously used to disable virtual pageviews in Google Analytics for OnTrack.
Since the latest gtag.js script has already been added to index.html which is compatible with Angular 17 and is working as expected. So, this legacy logic is no longer needed, and this cleanup helps simplify the codebase.

## Type of change

- Deleted: The analytics.coffee
- Modified: config.coffee (removed the reference of the anaytics.coffee)
- Modified: doubtfire-angularjs.module.ts (removed the reference of the anaytics.coffee)

# How Has This Been Tested?

To test this change, I first ran the project with the analytics.coffee file still in place. While navigating through different pages, I opened the Network tab in the browser’s developer tools and filtered the requests using the keyword “google” to track Google Analytics activity.

Then, I removed the analytics.coffee file and repeated the same steps. I compared both results to check for any differences in tracking behavior.

I also used the Google Analytics Debugger extension to inspect the presence and frequency of the "page_view" events. After comparing both runs, I didn’t observe any noticeable difference in tracking or event counts.

This confirmed that removing the analytics.coffee file does not impact the current analytics setup.

### Screenshots ###

Before removal of analytics.coffee
![with](https://github.com/user-attachments/assets/cd53d1e3-cb28-46b4-bb61-9a0f9d4903a8)



Testing with Google Analytics Debugger Tool : (before removal of analytics.coffee )
![before gradtool](https://github.com/user-attachments/assets/8c8d6ffa-98af-4154-a81d-24cf33e3d7c5)



After removal of analytics.coffee
![without](https://github.com/user-attachments/assets/8b2df90e-5e4e-4734-b270-2140c101113a)


Testing with Google Analytics Debugger Tool : (after removal of analytics.coffee )
![after-gadtool](https://github.com/user-attachments/assets/376f1c1e-432f-4399-b807-365adab93b92)




## Testing Checklist:

- [x] Tested in latest Chrome
- [ ] Tested in latest Safari
- [ ] Tested in latest Firefox

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have requested a review the Pull Request

zara1504 commented:
Hi there, after testing and looking over this PR I can see that you have removed the analytics.coffee file and removed it from other places in different files. Following your testing above and testing it myself everything went well for me. One thing I can say is that in the doubtfire-angularjs.module.ts file, where you removed the line of code, you have left a blank space within everything so I would just recommend removing that blank line so everything looks clean. Overall, everything is good. Good job. :)

disururathnayake commented:
After testing over and over again I found that there is no harm from the removal of analytics.coffee file. All the reference also got removed from other files. Everything's works fine, Good work.

atharv02-git reviewed (approved):
Hi Pasindu, went through your changes, i can tell that existence of analytics.coffee has no impact on our doubtfire-web, so removing it was the only option. You have unlinked all the references for analytics.coffee properly.
Your findings in the component review along with the way you approached, was very good.
Just one thing here, in future whenever you push to any branch don't push either package.json or package-lock.json in order to avoid conflicts.
LGTM! Keep up the good work

aNebula commented:
@Pasindufdo98  please resolve the conflicts

Pasindufdo98 commented:
> @Pasindufdo98 please resolve the conflicts

Hi @aNebula I removed the package.json and package-lock.json files which were included in a previous commit that was unnecessary for this migration PR.

atharv02-git commented:
> > @Pasindufdo98 please resolve the conflicts
> 
> Hi @aNebula I removed the package.json and package-lock.json files which were included in a previous commit that was unnecessary for this migration PR.

Can C

> > @Pasindufdo98 please resolve the conflicts
> 
> Hi @aNebula I removed the package.json and package-lock.json files which were included in a previous commit that was unnecessary for this migration PR.

Can now confirm author has removed package.json and package-lock.json

aNebula commented:
LGTM.
Please open the upstream pull request with these changes against [doubtfire-lms/doubtfire-web 9.x branch](https://github.com/doubtfire-lms/doubtfire-web/tree/9.x)

b0ink committed:
refactor: replace range datepickers with three separate datepickers

b0ink committed:
feat: display session start and end time

b0ink committed:
chore(release): 10.0.0-49

b0ink committed:
feat: jplag base code (#1010)

* feat: jplag base code

* chore: reword base code description

b0ink committed:
chore(release): 10.0.0-50

b0ink committed:
fix: unlock tasks when prerequisite requires rff and task is in aip state

b0ink committed:
chore(release): 10.0.0-51

b0ink committed:
chore: filter out sessions with less than 1 minute duration

b0ink committed:
fix: filter out students from unit staff editor (#1018)

b0ink committed:
refactor: send timezone of client to the api (#1019)

b0ink committed:
feat: download marking sessions for tutor

b0ink committed:
fix: display tutor name

b0ink committed:
refactor: expand hour segments ui

b0ink committed:
fix: re-enable user filtering on click

b0ink committed:
refactor: migrate group members list (#1017)

* refactor: init group members list migration

* refactor: migrate group members list

* chore: update loading text

* chore: remove old component files

b0ink committed:
refactor: collapse related tutorial list

b0ink committed:
feat: campus timezone (#1022)

b0ink committed:
Merge pull request #1020 from b0ink/feat/download-my-marking-sessions

feat: download my marking sessions

b0ink committed:
refactor: enable marking sessions calendar for tutors

b0ink committed:
chore(release): 10.0.0-52

b0ink committed:
refactor: include all submitted tasks in portfolio task list (#1023)

* refactor: include all submitted tasks in portfolio task list

* refactor: fetch tasks included in portfolio from api

* refactor: reword

* refactor: reword

* refactor: show warning if no tasks found

* refactor: wording

* refactor: add error alert

b0ink committed:
chore(release): 10.0.0-53

b0ink committed:
refactor: number list

b0ink committed:
chore: add margin

b0ink committed:
chore(release): 10.0.0-54

b0ink committed:
fix: typo

b0ink committed:
fix: display correct marking session details

b0ink committed:
feat: display task count in inbox (#1025)

* feat: display number of tasks in inbox

* chore: only display task count if not empty

b0ink committed:
refactor: scrollable calendar (#1027)

b0ink committed:
chore(release): 10.0.0-55

b0ink committed:
refactor: require comment for assess in portfolio tasks submitting for feedback (#1026)

* refactor: require comment for assess in portfolio tasks submitting for feedback

* chore: fix comment

* chore: fix indentation

* fix: condition

* chore: update character count text color

* chore: add tooltip

b0ink committed:
fix: ensure grade has been selected

b0ink committed:
chore: display error on failed grade change

b0ink committed:
chore(release): 10.0.0-56

returnMarcco committed:
docs: update frontend migration progress list (#982)

* docs: update frontend migration progress list

- Mark all components in the `thoth-tech/9.x` branch as `MIGRATED` if they exist as TS files.

- Mark components with open PRs against the `doubtfire-lms` repo as `MIGRATED`.

* docs: update frontend migration progress list

- remove completed migrations from TODO section

* docs: update frontend migration progress list

- Amend progress list

* docs: update frontend migration progress list

- resolve discrepancies in progress list

- add additional section for components `awaiting upstream review`

- add instructions for PR author and upstream repo maintainer

- update summary section

* docs: fix typos

- fix typos for components in progress list

* docs: amend frontend migration list changes

- remove `AWAITING UPSTREAM REVIEW` section as this wouldn't work for our workflow

- move components from the `AWAITING UPSTREAM REVIEW` section back to the `TODO` section

- leave comment to update `migration progress list` on all PRs for components that were previously in the `AWAITING UPSTREAM REVIEW` section. to verify this, search the PR section in `doubtfire-lms/doubtfire-web`

* docs: further updates to README.MD

- move migrated component to `MIGRATED` section of `Migration Progress List`

* docs: amend migration progress list

- mark components that have .ts files as being migrated

* docs: amend frontend migration progress list

- amend the `No longer in thoth-tech/9.x` section of the frontend migration progress list

* docs: remove duplicates & update tally

- remove duplicates from the MIGRATED section of the frontend migration progress list

- amend tally

* docs: fix formatting

- fix formatting of frontend migration progress list in README.MD

* chore: update repository reference

---------

Co-authored-by: Boink <40929320+b0ink@users.noreply.github.com>

b0ink committed:
fix: ensure selected group is valid

b0ink committed:
refactor: group selector migration (#1030)

* refactor: init group selector migration

* refactor: move group set selector

* chore: change color of locked icon

* refactor: ability to join group

* refactor: conditional rendering for project view

* chore: ensure group is selected

* chore: add information if group doesnt exist

* chore: disable click event while editing

* refactor: add filters

* chore: fix margins

* chore: fix margins

* chore: remove coffeescript files

* chore: cleanup

* refactor: add locked icon

* refactor: remove group set selector

b0ink committed:
refactor: use students preferred name in task comments (#1028)

* refactor: use students preferred name in task comments

* refactor: use users preferred name

b0ink committed:
fix: typo

b0ink committed:
chore: update task status from latest submission details (#1031)

b0ink committed:
fix: enable submission button for 'need help' trigger (#1034)

b0ink committed:
refactor: replace task definition editor with vertical stepper (#1032)

* refactor: replace task def editor with vertical stepper

* chore: capitalise labels

* fix: typo

* fix: typo

b0ink committed:
feat: open similarities from project dashboard (#1035)

* feat: open similarities from project dashboard

* chore: add task similarity view component

b0ink committed:
chore(release): 10.0.0-57

b0ink committed:
refactor: group set manager migration (#1036)

* refactor: init group set manager migration

* refactor: allow editable group name

* chore: display locked icon

* refactor: replace with new component

* refactor: shorten create group button

* refactor: remove old component

* chore: cleanup

* refactor: hide search bar for students

* refactor: remove css

b0ink committed:
chore: display joined status

b0ink committed:
chore: remove group members loaded callback

b0ink committed:
chore: null check

b0ink committed:
refactor: imply portfolio has been submitted (#1037)

b0ink committed:
chore(release): 10.0.0-58

b0ink committed:
chore: ensure student selects grade before confirming learning summary report

b0ink committed:
fix: require comment for new evidence

b0ink committed:
fix: require comment for new evidence

b0ink committed:
chore(release): 10.0.0-59

b0ink committed:
feat: upload grades csv (#1038)

* feat: upload grades csv

* chore: remove debug

b0ink committed:
chore(release): 10.0.0-60

b0ink committed:
refactor: overseer sidekiq (#1029)

* refactor: use sidekiq jobs for pulling overseer images

* refactor: overseer execution script editor

* chore: remove overseer view

* refactor: increase size of script editor

* chore: add icon for incomplete overseer runs

* chore: add typing

* refactor: format assessment comment layout

* refactor: enable test submissions if script exists

b0ink committed:
chore(release): 10.0.0-61

b0ink committed:
feat: restrict assessments to tutors in the same tutorial stream (#1033)

* feat: restrict assessments to tutors in the same tutorial stream

* refactor: unselect tasks that are locked to a tutorial stream

b0ink committed:
chore(release): 10.0.0-62

b0ink committed:
chore: replace doubtfire with ontrack

b0ink committed:
feat: open project dashboard from portfolios view

(#1040)

b0ink committed:
chore(release): 10.0.0-63

b0ink committed:
fix: avoid calling window to open project in new tab

b0ink committed:
refactor: move overseer script url to task definition model

b0ink committed:
refactor: base64 encode overseer script before sending to server

b0ink committed:
Merge pull request #1041 from b0ink/fix/base64-overseer

fix: base64 encode overseer script

b0ink committed:
fix: use new google fonts api for proper weight loading (#1039)

b0ink committed:
chore: specify only time exceeded tasks are updated

b0ink committed:
chore: disable no inferrable types rule

b0ink committed:
refactor: migrate file uploader (#1043)

* chore: init file uploader migration

* refactor: clean up ui and reset uploader after completion

* chore: use external name on error

* refactor: migrate remaining file uploaders

* refactor: center error size

* refactor: ensure uploader is hidden but remains in dom

* refactor: remove logs

* refactor: improve type safety

* chore: typo

b0ink committed:
chore: remove any types

b0ink committed:
chore: remove old file uploader component

b0ink committed:
chore: update migration progress

b0ink committed:
chore: remove unused files (#1044)

b0ink committed:
refactor: portfolio welcome step migration (#1045)

* refactor: init portfolio welcome step migration

* chore: remove old portfolio welcome step component

* chore: remove whitespace

* chore: add todo

* chore: increase padding

b0ink committed:
refactor: migrate portfolio learning summary report step (#1046)

* refactor: migrate portfolio learning summary report step

* fix: typo

* refactor: remove unused code

* chore: update placeholder

* refactor: check if draft task definition was submitted

* chore: add type

* refactor: add caption

* refactor: add warning to ensure justification

* chore: remove old learning summary report step component

* chore: unlink old learning summary report step component

* docs: update migration progress

* chore: add typing

b0ink committed:
refactor: migrate portfolio add extra files step component (#1047)

* refactor: migrate portfolio add extra files step component

* chore: cleanup whitespace

* chore: remove unused class

* chore: remove debug

* chore: remove old portfolio add extra files step component

* chore: update migration progress

b0ink committed:
refactor: unit portfolios state migration (#1048)

* chore: init unit portfolios state migration

* chore: init portfolios directives

* refactor: init tab view

* refactor: migrate select student portfolio list

* refactor: hook student selection

* chore: ensure pdf viewer takes up screen

* chore: init project progress dashboard

* chore: view progress on project selection

* refactor: impove project progress layout

* refactor: init portfolio assessment view

* refactor: portfolio list pagination

* fix: prevent vertical in tab groups

* refactor: add padding in child components

* refactor: add target and submitted grade card ui

* refactor: improve assessment layout

* refactor: add student search and filtering

* feat: add tab headings

* fix: set correct tutorial filter

* chore: change paginator options

* fix: distribute layout evenly

* chore: add loading text

* chore: remove todo

* feat: download grades button

* chore: add placeholders todo

* fix: update method call

* refactor: improve graph layout

* refactor: add route error handling

* chore: update todo

* chore: update todos

* feat: has portfolio column

* feat: view project in new tab

* chore: add todo

* refactor: display no portfolio if doesnt exist

* fix: typo

* chore: consistent gap

* chore: remove old portfolios component

* chore: update task stats on grade change

* chore: remove old project progress dashboard

* chore: display error alert

* chore: remove old project progress dashboard

* chore: update migration progress

* refactor: improve assessment scores responsiveness

b0ink committed:
chore: update migration progress

b0ink committed:
chore: remove admin modals coffeescript

b0ink committed:
chore: update migration progress

b0ink committed:
fix: display groups only when a group set is selected

b0ink committed:
chore: revert hidden condition

b0ink committed:
chore: update migration progress

b0ink committed:
chore: update migration progress

b0ink committed:
chore(release): 10.0.0-64

b0ink committed:
fix: typo

b0ink committed:
chore(release): 10.0.0-65

b0ink committed:
refactor: migrate rollover state (#1050)

* refactor: init rollover migration

* refactor: add rollover functionality

* refactor: remove comment

* chore: remove old rollover component

* chore: revert formatting

* chore: revert formatting

* chore: update migration progress

* chore: update migration progress

* chore: update migration progress

* chore: update migration progress

* chore: update migration progress

* chore: update migration progress

b0ink committed:
fix: correctly set rollover end date

b0ink committed:
refactor: migrate unit groups (#1049)

* refactor: migrate unit groups

* chore: update migration progress

* chore: update migration progress

* chore: add comment

b0ink committed:
refactor: rename chart components

b0ink committed:
chore: remove unused visualisations

b0ink committed:
chore: remove unused visualisations

b0ink committed:
chore: remove unused visualisations

b0ink committed:
feat: discussion prompts (#1042)

* feat: init discussion prompts ui

* refactor: improve tutor discussion ui

* refactor: display discussion prompts in tutor discussion view

* refactor: reword weight to priority

b0ink committed:
refactor: migrate project groups (#1056)

* refactor: migrate project groups component

* chore: ensure callback is not null

* chore: update migration progress

b0ink committed:
refactor: migrate group member contribution assigner component (#1057)

* feat: migrate group-member-contribution-assigner component

* refactor: use mat table

* refactor: add table sorting

* chore: update migration progress

* chore: avoid defaulting group

* fix: check if overseer test submission

* refactor: remove groups module

* refactor: improve hover rating effect

* refactor: use modern for-loop directive

* chore: default sort by student name

* refactor: ensure confrating is set

- used in other logic for statistics

* refactor: remove unused code

* refactor: remove confrating

* chore: allow nullable overstar property

* chore: remove class

---------

Co-authored-by: EkamBhullar <bhullarekam04@gmail.com>

b0ink committed:
refactor: require confirmation for deleting comment (#1059)

b0ink committed:
refactor: render overseer comments as is (#1058)

b0ink committed:
chore(release): 10.0.0-66

b0ink committed:
feat: allow sidekiq web access (#1053)

b0ink committed:
chore(release): 10.0.0-67

b0ink committed:
fix: only trim if valid

b0ink committed:
chore(release): 10.0.0-68

b0ink committed:
feat: display number of stuff notes in tutor discussion

b0ink committed:
feat: attention required task status (#1061)

* feat: init discuss check task status

* refactor: add discuss check status

* refactor: add discuss and check marking button

* refactor: add warning badge to discuss check tasks

* refactor: update discuss check status description

* refactor: update discuss check description

* chore: ensure discuss check status does not unlock requisite tasks

* refactor: rename to off track

* refactor: rename to attention required

* refactor: update status colors

* refactor: rename to attention required internally

* chore: update assess in portfolio color

b0ink committed:
feat: overseer pipeline (#1064)

* feat: init overseer pipeline ui

* chore: init drag and drop steps panel

* feat: init overseer-step crud

* feat: use new monaco editor package

* refactor: clean up ui

* feat: get list of overseer resource files

* chore: allow no input file

* feat: allow partial input output diff check

* feat: allow deleting of overseer steps

* feat: overseer step result model

* refactor: improve overseer report view

* refactor: organise imports

* refactor: retrieve feedback message

* refactor: expose overseer assessment details to task comment

* feat: dropdown for command language

* fix: set fixed height size

* chore: show overseer reports to students

* build: ensure monaco editor is packed correctly

* refactor: lazy load step results for overseer assessments

* refactor: align default feedback messages with the backend

* refactor: improve assessment comment ui

* chore: ignore old assessment script

* refactor: hide test result until report is ready

* fix: use new overseer report component for test submission history

* chore: allow for overseer report refresh

* refactor: reduce icon size

* refactor: auto expand selected overseer report

* chore: enable submission history if project is valid

* chore: delete new step

* chore: set correct overseer steps when changing task definition

* feat: add overseer task status result

* chore: add hint explaining test submissions can be used before enabling overseer

* refactor: hide overseer steps until docker image has been selected

* chore: move overseer editor to original place

* refactor: ensure task definition is saved before adding overseer steps

* chore: show delete and save buttons when step is selected

* refactor: use seconds for overseer timeout

* refactor: show status dropdown only when halting step

* refactor: cleanup

* chore: render diff if overseer step is null

* refactor: remove debug

* chore: show empty overseer reports

b0ink committed:
feat: project task planner gantt chart (#1051)

* feat: init project plan gatt chart

* chore: set compatible version

* chore: update gantt chart package

* refactor: display task items and map prerequisites

* refactor: add task target start and due date

* refactor: show prerequisite for tasks in tooltip

* refactor: save target and due dates

* feat: display task definition dates as baseline items

* chore: remove extension

* fix: create unique baseline item

* refactor: fix timezone issues

* refactor: clean up ui buttons

* refactor: ensure timeline view shows the earliest task

* chore: always show task planner

* refactor: rename project plan

* refactor: minify ui

* chore: confirm changes

* feat: allow target grade changes

* refactor: display tasks target grade

* feat: link to task details

* refactor: remove debug

* chore: remove old task sliders

* refactor: display show date toggle

* chore: add curved links

* fix: save target dates

* chore: reverse prerequisite links

* refactor: reset target dates if task is prerequisite for another task

* chore: fix gantt chart config

* refactor: display warning if prerequisite task ends after dependent task starts

* fix: return false if no linked item

* refactor: highlight prerequisite conflicts

* refactor: move task planner into its own component

* refactor: highlight prerequisites on task hover

* chore: fix prerequisite conflicts

* feat: add status icon to task list

* chore: update bar color

* feat: display task prerequisites modal

* refactor: display dependent tasks

* fix: only use target start and end date if unit allows flexible dates

* refactor: ensure prerequisites are mapped

* chore: only update z-index of lines when hovering over

* feat: task planner card for project dashboard

* chore: add more details

* chore: add tooltip delay

* chore: add description heading

* chore: add back watch command

* refactor: use single endpoint to reset all target dates

* chore: add button to redirect to task planner

* chore: only show dates in tooltip

* refactor: set gantt view dates to latest possible end date

* feat: scroll to and highlight selected task item

* refactor: remove unused code

* refactor: remove unused code

* chore: format

* refactor: assign gantt config variable

* refactor: add oop logic + cleanup duplicated code

* fix: convert correct deadline date string

* chore: import task planner card

b0ink committed:
chore(release): 10.0.0-69

b0ink committed:
chore: allow no expected output file

b0ink committed:
chore: render correct error

b0ink committed:
fix: skip query for unsaved task definition

b0ink committed:
chore: remove required fields

b0ink committed:
chore(release): 10.0.0-70

b0ink committed:
feat: download staff notes csv (#1062)

* feat: download staff notes csv

* chore: restore imports

b0ink committed:
feat: migrate to jplag report viewer v6.3.0 (#1082)

* feat: migrate to jplag report viewer v6.3.0

- now runs on vanilla JPlag report viewer

* chore: update jplag build

* chore: rename variable

b0ink committed:
chore(release): 10.0.0-71

b0ink committed:
fix: null safe user access

b0ink committed:
refactor: render sessions with missing staff

- tutors may have had marking sessions, but then removed from the unit

b0ink committed:
chore(release): 10.0.0-72

b0ink committed:
chore: default jplag src

b0ink committed:
refactor: attempt to scroll jplag comparisons until component is found

b0ink committed:
chore(release): 10.0.0-73

b0ink committed:
chore(release): 10.0.0

b0ink committed:
refactor: update task definition resource files (#1066)

b0ink committed:
chore(release): 10.0.1-0

b0ink committed:
refactor: disable target grade change from project dashboard (#1084)

b0ink committed:
refactor: improve wording and ux for assess in portfolio submissions (#1083)

* refactor: improve wording and ux for assess in portfolio submissions

* chore: increase max width

* refactor: expand on wording

b0ink committed:
feat: tutor notes + mentorship + moderation + escalation (#1068)

* feat: init tutor notes and mentorship

* feat: tutor moderation route

* feat: add moderator actions

* refactor: add confirmation modals

* refactor: move moderation buttons to footer

* feat: filter tutor notes by task

* refactor: mark tutor notes as read

* refactor: view tutor notes via modal

* fix: remove task definition filters for moderation

* chore: add dividers for moderation buttons

* refactor: add tooltip

* refactor: clean up tasks tutor getter

* refactor: rename vars to tutor notes

* refactor: add typing

* refactor: clean up unused code

* fix: check for valid tutor

* feat: init escalation request

* chore: add moderation type field

* refactor: add alternate ui for escalated tasks

* chore: add moderation action type

* feat: feedback appeal modal

* feat: add feedback review requested comment

* chore: revert task-submission-card

* feat: add moderation confirmation modal and allow dismiss all

* chore: dismiss modal

* feat: filter tasks by tutor

* refactor: style action buttons

* chore: rename staff to tutor

* refactor: improve wording

* refactor: improve styling

* chore: add relevant icon

* refactor: update success alert

* chore: increase alert duration

* chore: display tasks tutor

* refactor: sort tutors alphabetically

* chore: remove divider on the right

* refactor: require a comment for review requests

* chore: increase modal max width

b0ink committed:
fix: ensure tutor notes view extends full height

b0ink committed:
chore(release): 10.0.1-1

b0ink committed:
fix: set min width for mentor dropdown

b0ink committed:
refactor: show moderation link to mentors only

b0ink committed:
refactor: group tutor filter by mine and all tutors mentored

b0ink committed:
chore: always show moderation to convenors

b0ink committed:
chore: default to my tutors only if in moderation view

b0ink committed:
Merge pull request #1087 from b0ink/feat/moderation-filters

refactor: group tutor filter by mine and all tutors mentored

b0ink committed:
chore(release): 10.0.1-2

b0ink committed:
chore: disable prerequisite due date validation

b0ink committed:
Merge pull request #1088 from b0ink/chore/disable-prereq-date-validation

chore: disable prerequisite due date validation

b0ink committed:
chore(release): 10.0.1-3

b0ink committed:
feat: allow custom target dates per target grade (#1089)

* feat: allow custom target dates per target grade

* refactor: rename to target dates for clarity

* chore: rename to target date

* refactor: fade datepicker if inherited from default target date

* chore: toast erron on failed task update

* fix: only reset items that are rendered

* fix: ensure baseline items refresh on grade change

* chore: add page heading

* refactor: return per grade due dates if dates are flexible

* chore: toggle due date ui

b0ink committed:
chore: add link to task planner

b0ink committed:
chore: remove debug

b0ink committed:
chore(release): 10.0.1-4

b0ink committed:
feat: per grade start dates (#1090)

* feat: per grade start dates

* feat: display errors when target date is before start date

b0ink committed:
refactor: use projects task dates to get earliest start date

b0ink committed:
chore(release): 10.0.1-5

b0ink committed:
refactor: encode script content to avoid firewall conflicts (#1091)

b0ink committed:
chore(release): 10.0.1-6

b0ink committed:
fix: ensure timestamp is normalised to start of day

b0ink committed:
chore(release): 10.0.1-7

b0ink committed:
feat: overflow marking (#1086)

* feat: init overflow marking

* chore: show overflow link to only staff who can overflow mark

* feat: enable overflow marking permission saving

* feat: feedback overflow unit setting

* chore: track if task is loading submission details

* feat: claim task request

* chore: ensure feedback thresholds are saved

* refactor: disable buttons based on claim conditions

* chore: call claim task request

* fix: display message for task comment errors

* refactor: clean up unit role getter

* refactor: display overflow claim state

* chore: reword label

* chore: add error alert

* fix: check if key exists so it can be reset to null

* refactor: add tooltips, show claim button in box

* chore: remove todo

* refactor: add snackbar

* refactor: move task claim to its own component

* feat: display overdue feedback warnings in staff task list

* refactor: reword task claim tooltip

* chore: extend type

b0ink committed:
chore(release): 10.0.1-8

b0ink committed:
fix: typo

b0ink committed:
chore(release): 10.0.1-9

b0ink committed:
feat: display available storage for docker images (#1092)

* feat: display available storage for docker images

* refactor: hide available storage if disabled

b0ink committed:
chore(release): 10.0.1-10

b0ink committed:
chore: increase allowed failed attempts

- Increase poll delay and failure threshold (5) to improve blue-green Caddy switchover buffer

b0ink committed:
fix: ensure deadline is returned

b0ink committed:
chore: sort task planner by start date

b0ink committed:
chore(release): 10.0.1-11

b0ink committed:
refactor: parse submission date with timestamp

b0ink committed:
chore(release): 10.0.1-12

b0ink committed:
feat: add snooze moderation action (#1096)

b0ink committed:
chore(release): 10.0.1-13

b0ink committed:
refactor: remove streamless tutorials (#1093)

* refactor: remove ui to add tutorial with no stream

* refactor: improve tutorial ui

* refactor: add ui to show no tutorials added

b0ink committed:
chore: add tooltip

b0ink committed:
chore: extend tutor notes max width

b0ink committed:
feat: view tutor notes from unit staff editor

b0ink committed:
chore(release): 10.0.1-14

b0ink committed:
chore: ensure all tasks filter is selected by default if task is null

b0ink committed:
chore: view tutor notes from task dropdown menu

b0ink committed:
chore(release): 10.0.1-15

b0ink committed:
chore: require confirmation for removing staff

b0ink committed:
chore(release): 10.0.1-16

b0ink committed:
fix: empty prompts layout

b0ink committed:
chore: always show mark as read button (#1098)

b0ink committed:
chore(release): 10.0.1-17

b0ink committed:
refactor: migrate unit group set editor (#1100)

* chore: init unit group set editor migration

* refactor: migrate unit group set editor

* chore: remove container class

* fix: set correct unit role

* chore: use mat button

* refactor: format

* chore: improve layout of group set manager

* chore: remove period

* fix: allow unlimited space if group set capacity is null

* chore: remove coffeescript files

b0ink committed:
refactor: migrate csv result + upload modal (#1099)

* chore: init csv result modal migration

* refactor: migrate csv result modal

* refactor: improve ui ux

* refactor: migrate csv-upload-modal

* chore: remove csv result and upload modal coffeescript files

* chore: remove comment

b0ink committed:
fix: dont subtract tutor note count for reading your own notes

b0ink committed:
fix: convert emojis in submission comment to string with colons (#1105)

b0ink committed:
chore(release): 10.0.1-18

b0ink committed:
chore: hide tutor notes icon in footer from tutors

b0ink committed:
feat: view submission history + zip file editor (#1104)

* feat: zip file viewer

* refactor: display upload requirement names instead of zipped file names if available

* feat: show file comparisons when selecting two submissions

* chore: ensure single editor sets correct language

* chore: close menu when selecting for comparison

* refactor: add task and student to modal title

* refactor: clean up ui

* fix: set correct submission number

* refactor: reduce duplicated code

* feat: add tree navigation and edit overseer resources

* refactor: show submission id and most recent

* chore: fix icon size

* refactor: clean up logic

* chore: view submission files as staff only

b0ink committed:
chore(release): 10.0.1-19

b0ink committed:
refactor: declutter footer ui (#1106)

* refactor: declutter footer ui

- toggle between moderation and status buttons in moderation view
- move discussed in class button to extra actions menu
- swap complete and discuss buttons

* chore: remove old discussed in class

* chore: remove old status button switch

b0ink committed:
chore(release): 10.0.1-20

b0ink committed:
refactor: task definition editor refactor (#1063)

* refactor: init task definition editor refactor

* refactor: improve task definition editor

* refactor: improve task definition editor ui

* refactor: add back task definition editor buttons

* chore: highlight selected task definition

* chore: move add task button to right side

* chore: keep task badge centered

* chore: center task badge with arrow icon

* chore: add scroll buffer

b0ink committed:
chore(release): 10.0.1-21

b0ink committed:
feat: require discussion before marking complete (#1103)

* feat: require discussion before marking complete

* chore: wording

b0ink committed:
refactor: migrate project dashboard (#1052)

* refactor: progress unit tasks list ui

* chore: fix task list styling

* feat: load task details from url

* refactor: load task details

* refactor: add task comments viewer

* chore: fix tas list panel resize handle

* chore: add todo

* chore: ensure burndown chart is refreshed

* refactor: ensure progress burndown chart extends full height

* refactor: ensure task list updates on target grade change

* refactor: add submission options menu

* feat: inbox dashboard

* chore: use original inbox icon

b0ink committed:
refactor: migrate students list component (#1107)

* refactor: migrate students list component

* chore: import students list controller

* refactor: remove section div

b0ink committed:
Merge branch '9.x' into 11.0.x

b0ink committed:
chore: fix tasks state

b0ink committed:
refactor: migrate upload submission modal

b0ink committed:
Merge pull request #1108 from b0ink/11.0.x

Merge 9.x migrations into 11.0.x (10.0.x base)

b0ink committed:
refactor: remove unused coffeescript files

b0ink committed:
chore: remove portfolio tasks step component

b0ink committed:
Merge pull request #1109 from b0ink/refactor/remove-unused-angularjs

refactor: remove unused coffeescript files

b0ink committed:
chore: used migrated portfolio welcome step component

b0ink committed:
refactor: migrate portfolio review step component

b0ink committed:
chore: rename variable for clarity

b0ink committed:
Merge pull request #1110 from b0ink/refactor/migrate-portfolio-review-step

refactor: migrate portfolio review step component

b0ink committed:
chore: remove analytics service import

b0ink committed:
refactor: migrate portfolio creation state

b0ink committed:
chore: update README

b0ink committed:
Merge pull request #1111 from b0ink/refactor/portfolio-creation-state

refactor: migrate portfolio creation state

b0ink committed:
fix: set task data for project dashboard state

b0ink committed:
chore: direct to new project dashboard state

b0ink committed:
refactor: add task planner and unit ilos card

b0ink committed:
refactor: clean up unit task list ui

b0ink committed:
chore: scrollable project dashboard

b0ink committed:
fix: remove hardcoded chart view size

b0ink committed:
refactor: migrate project groups state

b0ink committed:
chore: set view type in projects parent state

b0ink committed:
Merge pull request #1112 from b0ink/refactor/migrate-project-groups-state

refactor: migrate project groups state

b0ink committed:
refactor: move tutorials state under new projects parent state

b0ink committed:
refactor: move task planner state to new projects parent state

b0ink committed:
chore: remove unused import

b0ink committed:
refactor: migrate timeout state

b0ink committed:
fix: avoid rendering the staff list twice

- prevents inbox queries from being called twice

b0ink committed:
Merge branch '10.0.x' into 11.0.x

b0ink committed:
Merge pull request #1113 from b0ink/refactor/migrate-timeout

refactor: migrate timeout state

b0ink committed:
Merge branch '11.0.x' of https://github.com/doubtfire-lms/doubtfire-web into 11.0.x

b0ink committed:
chore: remove errors module import

b0ink committed:
refactor: migrate inbox states (#1114)

* refactor: migrate inbox state

* chore: add unit task inbox state

* fix: ensure hard refresh doesnt redirect

* refactor: improve inbox state abstraction

* chore: remove inbox coffeescript states

b0ink committed:
refactor: migrate students list state

b0ink committed:
Merge pull request #1115 from b0ink/refactor/migrate-students-list-state

refactor: migrate students list state

b0ink committed:
chore: update migration progress

b0ink committed:
refactor: improve portfolios ui layout

b0ink committed:
chore: move portfolios state to new unit parent state

b0ink committed:
refactor: migrate unit groups state

b0ink committed:
Merge pull request #1116 from b0ink/refactor/migrate-unit-groups

refactor: migrate unit groups state

b0ink committed:
refactor: migrate analytics state (#1118)

* refactor: migrate analytics state

* chore: wrap component in container

* chore: reset indent

* chore: update README

b0ink committed:
chore: update migration progress

b0ink committed:
chore: remove unused feedback component

b0ink committed:
chore: add staff notes view to inbox

b0ink committed:
chore: null safe access

b0ink committed:
refactor: migrate unit admin editor state

b0ink committed:
chore: remove container class

b0ink committed:
Merge pull request #1119 from b0ink/refactor/migrate-unit-editor

refactor: migrate unit admin editor state

b0ink committed:
feat: staff note and similarity indicators

b0ink committed:
chore: remove content-editable angularjs component

b0ink committed:
chore: remove unused outcome-service angularjs component

b0ink committed:
chore: deprecate filters.coffee

b0ink committed:
Merge branch '11.0.x' into migrate/analytics-coffee
